linode/manager
# feat: [M3-7345] – Unrecommended Linode configuration indicators on VPC Detail page

## Description 📝
Assigning a linode to a VPC but not having the VPC interface be `primary` results in situations where network connectivity issues can arise (see ticket for more details). This PR adds warning indicators on the VPC Detail page when such configurations exist: a warning icon at the `SubnetLinodeRow` level

## Changes  🔄
- Interactive tooltip w/ warning icon added to SubnetLinodeRow when a linode has an unrecommended configuration
- `hasUnrecommendedConfiguration` utility function
- Test updates

## Preview 📷
<details>
<summary>Warning icon</summary>

![Screenshot 2023-11-21 at 4 04 26 PM](https://github.com/linode/manager/assets/114682940/5b004fcf-ce07-4f73-b443-f5dfe02e39dd)

</details>

## How to test 🧪

### Prerequisites
- An account with proper VPC tags
- A VPC with:
  - a subnet within it
  - at least one linode assigned to it, with the linode having:
    - a public interface in eth0, designated as the primary interface (you can do this in the Linode Edit Config dialog)
    - a VPC interface in eth1 

### Verification steps 
With the prereqs described above, observe:
- The warning icon being present for the linode with an unrecommended configuration

If you go to the linode with an unrecommended configuration and adjust it by making the VPC interface (eth1) the primary interface, returning to the VPC Detail page should show that the warning icon is no longer present for that linode.

## As an Author I have considered 🤔
